### PR TITLE
Adding some activation functions for RBFN

### PR DESCRIPTION
**Pull request by himanshupathak21061998**
Related to  #2261 
I am trying to add some non-linear activation functions for rbf network in this PR.

### CODE REVIEW

File: src/mlpack/methods/ann/activation_functions/inverse_quadratic_function.hpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Inverse Quadratic function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = x^2 \\
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/inverse_quadratic_function.hpp:24]: Do you mean` f(x)^-1 =` ?

File: src/mlpack/methods/ann/activation_functions/inverse_quadratic_function.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ public:
+  /**
+   * Computes the Inverse Quadratic function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  static double Fn(const double x)
+  {
+    return 1 /( 1 + x * x);
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/inverse_quadratic_function.hpp:39]:
```suggestion
    return 1  / ( 1 + x * x);
```

File: src/mlpack/methods/ann/activation_functions/inverse_quadratic_function.hpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+   * @param y The resulting output activation.
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Fn(const InputVecType& x, OutputVecType& y)
+  {
+    y = 1 /(1 + arma::pow(x, 2));
+  }
+
+  /**
+   * Computes the first derivative of the Spline function.
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/inverse_quadratic_function.hpp:55]: Now I'm confused, spline function, I thought this is the inverse quadratic function, maybe I missed something?

Comment (himanshupathak21061998) [reply src/mlpack/methods/ann/activation_functions/inverse_quadratic_function.hpp]: Sorry I just forgot to change docs here.

File: src/mlpack/methods/ann/activation_functions/multi_quadratic_function.hpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * The Multi Quadratic function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = √(1 + x^2) \\
+ * f'(x) = x/f(x) \\
+ * @f}
+ */
+class MultiquadFunction
```
Comment (iamshnoo) [inline src/mlpack/methods/ann/activation_functions/multi_quadratic_function.hpp:28]: The Inverse Quadratic function is named as ```InvQuadFunction``` , whereas this is named as ```MultiquadFunction```. 
I think we should stick to either ```q``` or ```Q``` for the word quad in the naming of the functions.
Personally, I like ```Q``` better.

File: src/mlpack/tests/activation_functions_test.cpp [hunk 5]
```diff
@@ -935,4 +940,104 @@ BOOST_AUTO_TEST_CASE(CELUFunctionTest)
...
+  CheckDerivativeCorrect<InvQuadFunction>(desiredActivations,
+                                        desiredDerivatives);
+}
+
+/**
+ * Basic test of the quadratic function.
+ */
+BOOST_AUTO_TEST_CASE(QuadraticFunctionTest)
+{
+  // Hand-calculated values using Python interpreter.
```
Comment (iamshnoo) [inline src/mlpack/tests/activation_functions_test.cpp:967]: The comments in the other test cases you added don't have the part of the sentence ```using Python interpreter```. Maybe we can skip this part for this test case as well?

File: src/mlpack/methods/ann/activation_functions/multi_quadratic_function.hpp (hunk 4 above)
Comment (himanshupathak21061998) [reply src/mlpack/methods/ann/activation_functions/multi_quadratic_function.hpp]: Yeah, I also like `Q`.

File: src/mlpack/tests/activation_functions_test.cpp [hunk 6]
```diff
@@ -935,4 +940,104 @@ BOOST_AUTO_TEST_CASE(CELUFunctionTest)
...
+                                         9.95913e-05 0.5 0.5 \
+                                         0.2 1");
+
+  const arma::colvec desiredDerivatives("-0.369822 -0.175152 -0.0937021 \
+                                         -0.000199183 -0.64 -0.64 -0.369822\
+                                         -0.5");
+
+  CheckActivationCorrect<InvQuadFunction>(activationData, desiredActivations);
+  CheckDerivativeCorrect<InvQuadFunction>(desiredActivations,
+                                        desiredDerivatives);
```
Comment (zoq) [inline src/mlpack/tests/activation_functions_test.cpp:959]: The indentation is slightly off.

File: src/mlpack/tests/activation_functions_test.cpp [hunk 7]
```diff
@@ -935,4 +940,104 @@ BOOST_AUTO_TEST_CASE(CELUFunctionTest)
...
+                                         46355.9 0.693147 0.693147 \
+                                         4.39445 0");
+
+  const arma::colvec desiredDerivatives("18.3923 94.6819 280.03866 \
+                                         1042462.1078 1.0137702 1.0137702 \
+                                         18.3923 0");
+
+  CheckActivationCorrect<SplineFunction>(activationData1, desiredActivations);
+  CheckDerivativeCorrect<SplineFunction>(desiredActivations,
+                                          desiredDerivatives);
```
Comment (zoq) [inline src/mlpack/tests/activation_functions_test.cpp:999]: See comment above.

File: src/mlpack/methods/ann/activation_functions/spline_function.hpp [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Spline function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = x^2 * log(1 + x) \\
+ * f'(x) = 2*x * log(1 + x) + x^2/(1 + x)\\
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/spline_function.hpp:25]:
```suggestion
 * f'(x) = 2 * x * log(1 + x) + x^2 / (1 + x)\\
```

File: src/mlpack/methods/ann/activation_functions/quadratic_function.hpp [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * The Quadratic function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = x^2 \\
+ * f'(x) = 2*x \\
+ * @f}
+ */
+class QuadFunction
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/quadratic_function.hpp:28]: Not sure should we call this `QuadraticFunction`?

File: src/mlpack/methods/ann/activation_functions/poisson1_function.hpp [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * The Poisson one function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = (x - 1) * e^-x \\
+ * f'(x) = e^-x + (1 - x) * e^-x \\
+ * @f}
+ */
+class Poisson1Function
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/poisson1_function.hpp:28]: Is there a second Poisson function?

File: src/mlpack/methods/ann/activation_functions/quadratic_function.hpp (hunk 9 above)
Comment (himanshupathak21061998) [reply src/mlpack/methods/ann/activation_functions/quadratic_function.hpp]: Done.

File: src/mlpack/methods/ann/activation_functions/poisson1_function.hpp (hunk 10 above)
Comment (himanshupathak21061998) [reply src/mlpack/methods/ann/activation_functions/poisson1_function.hpp:28]: Yes, there is Poisson 2 function also

Comment (zoq) [reply src/mlpack/methods/ann/activation_functions/poisson1_function.hpp:28]: Ahh, okay, thanks for the clarification.

File: src/mlpack/methods/ann/activation_functions/multi_quadratic_function.hpp [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Multi Quadratic function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = √(1 + x^2) \\
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/multi_quadratic_function.hpp:24]: I'm not sure doxygen can render that in a latex environment, so maybe we should use `\sqrt` here.

Comment (himanshupathak21061998) [reply src/mlpack/methods/ann/activation_functions/multi_quadratic_function.hpp]: Thanks for reviewing.

**Comment by zoq** [comment]
Thanks for putting this together.

**Comment by himanshupathak21061998** [comment]
Thanks for merging.

**Pull request merged**